ocaml-lsp-server/src/dune.ml

```diff
@@ -0,0 +1,45 @@
+open Import
+
+type state =
+  | Absent
+  | Out_of_date
+  | Running
+
+type t =
+  { scheduler : Scheduler.t
+  ; state : state Fiber.Ivar.t
+  }
+
+let create scheduler = { scheduler; state = Fiber.Ivar.create () }
+
+let state t : state Fiber.t =
+  let open Fiber.O in
+  let* state = Fiber.Ivar.peek t.state in
+  match state with
+  | Some s -> Fiber.return s
+  | None -> (
+    match Bin.which "dune" with
+    | None ->
+      let+ () = Fiber.Ivar.fill t.state Absent in
+      Absent
+    | Some bin ->
+      let bin = Fpath.to_string bin in
+      let dev_null () = Unix.openfile "/dev/null" [ Unix.O_RDWR ] 0o666 in
+      let stdin = dev_null () in
```

Comment:
Did you want to leave out the ability to send requests and receive responses from this PR?